server/controllers/FollowController.js

```diff
@@ -0,0 +1,227 @@
+import db from '../database/models';
+
+const { User, Follow } = db;
+
+/**
+ * Class representing FollowController
+ */
+class FollowController {
+  /**
+   * Handles the Follow Feature
+   * @param {object} req - Request object
+   * @param {object} res - Response object
+   * @param {*} next - Next function
+   * @returns {object} An object containing all the data related to the followed user
+   */
+  static followUser(req, res, next) {
+    const { username } = req.params;
+    User.find({
+      where: {
+        username
+      }
+    }).then((user) => {
+      if (user) {
+        const follow = user;
+        const { userId } = req.decoded;
+        if (userId !== follow.id) {
+          Follow.findOrCreate({
+            where: {
+              userId,
+              followId: follow.id
+            }
+          }).spread((check, created) => {
+            if (!created) {
+              return res.status(409).json({
+                message: `You are already following ${username}`
+              });
+            }
+            res.status(200).json({
+              message: `You are now following ${username}`
+            });
+          });
+        } else {
+          return res.status(409).json({
+            message: 'You cannot follow yourself'
+          });
+        }
+      } else {
+        return res.status(404).json({
+          message: 'User not found'
+        });
+      }
+    }).catch(next);
+  }
+
+  /**
+   * @function unfollow
+   * @summary Handles the unfollow feature
+   * @param {object} req - Request object
+   * @param {object} res - Response object
+   * @param {*} next - Incase of errors
+   * @returns {object} An object containing all the data related to the unfollowed user
+   */
+  static unfollowUser(req, res, next) {
+    const { username } = req.params;
+    User.find({
+      where: {
+        username
+      }
+    }).then((user) => {
+      if (user) {
+        const follow = user;
+        const { userId } = req.decoded;
+        Follow.destroy({
+          where: {
+            userId,
+            followId: follow.id
+          }
+        }).then((unfollow) => {
+          if (unfollow) {
+            res.status(200).json({
+              message: `You have Unfollowed ${username}`
+            });
+          } else {
+            res.status(404).json({
+              message: `You never followed ${username}`
+            });
+          }
+        }).catch(next);
+      }
+    });
+  }
+
+  /**
+     * @function getAllUserIFollow
+     * @summary Handles getting all users a logged in user is following
+     * @param {object} req - Request object
+     * @param {object} res - Response object
+     * @param {*} next - Incase of errors
+     * @returns {Array} An array of object containing required user data
+     */
+  static getAllUserIFollow(req, res, next) {
+    const { username } = req.params;
```

Comment:
Bro, Do you know that the way you implemented this feature can break the app? Please do it again.

Reply:
Thank you for the feedback, Kingsley. But please can you be specific on what exactly can break the app?